src/NServiceBus.Core/CriticalError/CriticalError.cs

```diff
@@ -49,17 +56,43 @@ public virtual void Raise(string errorMessage, Exception exception)
             Guard.AgainstNull(nameof(exception), exception);
             LogManager.GetLogger("NServiceBus").Fatal(errorMessage, exception);
 
-            if (Endpoint == null)
+            lock (endpointCriticalLock)
             {
-                throw new InvalidOperationException("You can only raise critical errors on a started endpoint but the endpoint wasn't started yet.");
+                if (endpoint == null || criticalErrors.Any())
```

Comment:
I can't come up with a scenario where endpoint is not null but criticalErrors still contains something, that's what the lock is for?